en/02_Developer_Guides/03_Forms/Field_types/04_GridField.md

```diff
@@ -315,48 +320,59 @@ $form->setFields(FieldList::create(
 
 ### many_many_extraFields
 
-The component also has the ability to load and save data stored on join tables when two records are related via a 
-"many_many" relationship, as defined through [DataObject::$many_many_extraFields](api:SilverStripe\ORM\DataObject::$many_many_extraFields). While loading and saving works 
-transparently, you need to add the necessary fields manually, they're not included in the `getCMSFields()` scaffolding.
+The component also has the ability to load and save data stored on join tables when two records are related via a
+"many_many" relationship, as defined through `$many_many_extraFields` configuration (see [many_many relationships](/developer_guides/model/relations/#many-many)).
+While loading and saving works transparently, you need to add the necessary fields manually.
 
-These extra fields act like usual form fields, but need to be "namespaced" in order for the `GridField` logic to detect 
-them as fields for relation extra data, and to avoid clashes with the other form fields.
+These fields aren't included in the `getCMSFields()` scaffolding, and you shouldn't include them in the `getCMSFields()` implementation
+for the model being displayed in the `GridField` since that model could be used in other contexts as well where the extra fields
+aren't available.
 
-The namespace notation is `ManyMany[<extradata-field-name>]`, so for example `ManyMany[MyExtraField]`.
+These extra fields act like usual form fields when displayed in the context of a `GridField`.
+For the field list being passed into the `GridFieldDetailForm` you need to "namespace"
+the fields in order for the field value to be saved, and to avoid clashes with the other form fields.
 
+If you are only using this in a ready-only context and you know you don't have other fields with the same name on
```

Comment:
Things change over time so maybe it's bad practice to not namespace things even though it makes sense at that particular moment?

Reply:
I don't have strong feeling either way so I'll remove this sentence.